modules/ROOT/pages/errors/gql-errors.adoc

```diff
@@ -0,0 +1,107 @@
+:description: This section describes the GQLSTATUS errors that Neo4j can return, grouped by category, and an example of when they can occur.
+
+[[neo4j-gqlstatus-errors]]
+= List of all GQLSTATUS server error codes
+
+The following page provides an overview of all server errors in Neo4j, along with some scenarios and their possible solutions.
+All errors in Neo4j have severity level `ERROR`.
+
+* <<_connection-exceptions, Connection exceptions>>
+
+
+[[_connection-exceptions]]
+== Connection exceptions
+
+The following is a list of the connection exceptions that can occur in Neo4j.
+
+=== 08000
+
+Notifies the client that an exception occurred during the connection.
+
+.Error details
+[cols="<1s,<4"]
+|===
+|GQLSTATUS code
+m|0800
+|Status description
+a|error: connection exception
+|Classification
+m|CLIENT_ERROR
+|===
+
+=== 08007
+
+Notifies the client that an exception occurred during the resolution of a transaction.
+
+.Error details
+[cols="<1s,<4"]
+|===
+|GQLSTATUS code
+m|08007
+|Status description
+|error: connection exception - transaction resolution unknown.
+|Classification
+m|CLIENT_ERROR
+|===
+
+=== 08N00
+
+Notifies the client that an exception occurred while trying to connect to a specified database.
+
+.Error details
+[cols="<1s,<4"]
+|===
+|GQLSTATUS code
+m|08N00
+|Status description
+a|error: connection exception - unable to connect to database. Unable to connect to database `$db`. Unable to get bolt address of the leader. Check the status of the database. Retrying your request at a later time may succeed.
+|Classification
+m|CLIENT_ERROR
+|===
+
+=== 08N01
+
+Notifies the client that an exception occurred while trying to write to a specified database.
+
+.Error details
+[cols="<1s,<4"]
+|===
+|GQLSTATUS code
+m|08N01
+|Status description
+a|error: connection exception - unable to write to database. Unable to write to database `$db` on this server. Server-side routing is disabled. Either connect to the database leader directly or enable server-side routing by setting `$cfgSetting=true`.
+|Classification
+m|CLIENT_ERROR
+|===
+
+=== 08N02
+
+Notifies the client that an exception occurred while trying to connect to a specified database.
+
+.Error details
+[cols="<1s,<4"]
+|===
+|GQLSTATUS code
+m|08N02
+|Status description
+a|error: connection exception - unable to route to database. Unable to connect to database `$db`. Server-side routing is disabled. Either connect to `$db` directly, or enable server-side routing by setting `$cfgSetting=true`.
+|Classification
+m|CLIENT_ERROR
+|===
+
+=== 08N03
+
+Notifies the client that an exception occurred while trying to connect to a specified database.
```

Comment:
We can investigate whether this is correctly assigned, but I think it should be part of a different task. Here, the PR is only about adding the codes from the spreadsheet.

Reply:
@Lojjs, could you please look at this and see if this is a connection or a permission problem?

Reply:
This is used as top level error with a more specific cause by Cypher composite.

The old error message where it was used is "Writing in read access mode not allowed. Attempted write to $graph" and "Writing to more than one database per transaction is not allowed. Attempted write to $db1, currently writing to $db2", so I suppose it is when you do write to the database rather than connect to it, but it is not exactly a permission problem. If you want to know more details I would refer you to Nadja or Linnea in the Cypher composite team for this one.